File: Entities/ProblemIdea.cs
using System.Collections.Generic;

namespace MindYourMoodWeb.Entities
{
    public class ProblemIdea : Entity
    {
        //public int Id { get; set; }
        public ProblemStep Step { get; set; }
        public Problem Problem { get; set; }
        public string IdeaText { get; set; }
        public ICollection<ProblemProCon> ProsAndCons { get; set; }
    }
}

File: ClientApp/src/app/_model/problemIdea.ts
import { ProblemStep } from './problemStep';
import { ProblemProCon } from './problemProCon';
import { Problem } from './problem';

export interface ProblemIdea {
  id: number;
  step: ProblemStep;
  problem: Problem;
  ideaText: string;
  prosAndCons: ProblemProCon[];
}
